InventoryLowQuantityNotification/Helper/NotifyStockQtyFilter.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryLowQuantityNotification\Helper;
+
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryLowQuantityNotificationApi\Api\Data\SourceItemConfigurationInterface;
+
+/**
+ * class NotifyStockQtyFilter
+ *
+ * Notify Stock Quantity Filter Collection Based On notifyStockQty
+ */
+class NotifyStockQtyFilter
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var StockConfigurationInterface
+     */
+    private $stockConfiguration;
+
+    /**
+     * NotifyStockQtyFilter constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param StockConfigurationInterface $stockConfiguration
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        StockConfigurationInterface $stockConfiguration
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->stockConfiguration = $stockConfiguration;
+    }
+
+    /**
+     * Filter collection based on notifyStockQty
+     *
+     * @param array $data
+     * @param int $storeId
+     * @return void
+     */
+    public function execute(array $data, int $storeId): void
+    {
+        $notifyStockQtyConfigValue =  (int) $this->stockConfiguration
+            ->getNotifyStockQty($storeId);
+        array_filter(
```

Comment:
Is it possible to make some queries before and join some additional queries to the collection instead of filtering results? With the current approach it will cause performance issue with large amount of records when we will query them all and process on the PHP side. I prefer complex SQL query instead of post-processing.

Reply:
I think We may not required this class anymore as it is already been filtered inside the Grid class within the collection. I removed this class from implementation.